storage/allocator.go

```diff
@@ -224,44 +204,22 @@ func (a Allocator) RemoveTarget(existing []roachpb.ReplicaDescriptor) (roachpb.R
 	}
 
 	// Retrieve store descriptors for the provided replicas from the StorePool.
-	type replStore struct {
-		repl  roachpb.ReplicaDescriptor
-		store *roachpb.StoreDescriptor
-	}
-	replStores := make([]replStore, len(existing))
-	usedStat := stat{}
+	sl := new(StoreList)
 	for i := range existing {
 		desc := a.storePool.getStoreDescriptor(existing[i].StoreID)
 		if desc == nil {
 			continue
 		}
-		replStores[i] = replStore{
-			repl:  existing[i],
-			store: desc,
-		}
-		usedStat.update(desc.Capacity.FractionUsed())
+		sl.add(desc)
 	}
 
-	// Based on store statistics, determine which replica is the "worst" and
-	// thus should be removed.
-	var worst replStore
-	for i, rs := range replStores {
-		if i == 0 {
-			worst = rs
-			continue
-		}
-
-		if usedStat.mean < minFractionUsedThreshold {
-			if rs.store.Capacity.RangeCount > worst.store.Capacity.RangeCount {
-				worst = rs
-			}
-			continue
-		}
-		if rs.store.Capacity.FractionUsed() > worst.store.Capacity.FractionUsed() {
-			worst = rs
+	_, bad := a.balancer.selectBad(sl)
```

Comment:
Don't we need to check the `ok` return value here? Or if `selectBad` guarantees to return a non-nil pointer in all cases where there are existing nodes, this should be documented on the interface.